components/bl-dashlet-component/src/index.js

```diff
@@ -0,0 +1,149 @@
+import { useRef, useState, useCallback, useEffect } from 'react';
+import { ResizableBox } from './react-resizable.min';
+import { Dashlet } from './dashlet';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function DashletComponent({ component, eventHandlers, pods, instanceId }) {
+  const {
+    display, classList, style, title, height, width, resizing,
+    contextBlock, minWidth, maxWidth, minHeight, maxHeight, dragging
+  } = component;
+  const { contextBlockHandler } = eventHandlers;
+  const dashletContentPod = pods['dashletContent'];
+
+  const rootRef = useRef();
+
+  const [resizeMaxWidth, setResizeMaxWidth] = useState(maxWidth);
+  const [resizeMaxHeight, setResizeMaxHeight] = useState(maxHeight);
+
+  const [isOpen, setIsOpen] = useState(() => {
+    if (localStorage.getItem(instanceId + ' isOpen')) {
+      return localStorage.getItem(instanceId + ' isOpen') === 'true';
+    }
+
+    return true;
+  });
+
+  const [position, setPosition] = useState(() => {
+    if (localStorage.getItem(instanceId + ' position')) {
+      return JSON.parse(localStorage.getItem(instanceId + ' position'));
+    }
+
+    return { x: 0, y: 0 };
+  });
+
+  const [size, setSize] = useState(() => {
+    if (localStorage.getItem(instanceId + ' size')) {
+      return JSON.parse(localStorage.getItem(instanceId + ' size'));
+    }
+
+    return { height: height, width: width };
+  });
+
+  component.setSize = (size) => {
+    setSize(size);
+  };
+
+  component.getSize = () => {
+    return size;
+  };
+
+  component.setPosition = (position) => {
+    setPosition(position);
+  };
+
+  component.getPosition = () => {
+    return position;
+  };
+
+  component.setIsOpen = (isOpen) => {
+    setIsOpen(isOpen);
+  };
+
+  component.getIsOpen = () => {
+    return isOpen;
+  };
+
+  useEffect(() => {
+    localStorage.setItem(instanceId + ' position', JSON.stringify(position));
+  }, [position]);
+
+  useEffect(() => {
+    localStorage.setItem(instanceId + ' size', JSON.stringify(size));
+  }, [size]);
+
+  useEffect(() => {
+    localStorage.setItem(instanceId + ' isOpen', String(isOpen));
+  }, [isOpen]);
+
+  const onResizeStop = useCallback((e, data) => {
+    setResizeMaxWidth(maxWidth);
+    setResizeMaxHeight(maxHeight);
+    setSize({ width: data.size.width, height: data.size.height });
+  }, []);
+
+  const onResize = useCallback((e, data) => {
+    const parentElementOffsetLeft = rootRef.current.parentElement.getBoundingClientRect().left + window.pageXOffset;
```

Comment:
`rootRef.current.parentElement.getBoundingClientRect()` is called 4 times.

`const {top,left, ...} = rootRef.current.parentElement.getBoundingClientRect()`